src/transformers/modeling_tf_bert.py

```diff
@@ -833,84 +965,155 @@ def call(self, inputs, **kwargs):
-class TFBertForMaskedLM(TFBertPreTrainedModel, TFMaskedLanguageModelingLoss):
-
+class TFBertLMHeadModel(TFBertPreTrainedModel, TFCausalLanguageModelingLoss):
     authorized_unexpected_keys = [r"pooler"]
-    authorized_missing_keys = [r"pooler"]
+    authorized_missing_keys = [r"position_ids", r"predictions/decoder/bias"]
 
     def __init__(self, config, *inputs, **kwargs):
         super().__init__(config, *inputs, **kwargs)
 
-        if config.is_decoder:
-            logger.warning(
-                "If you want to use `TFBertForMaskedLM` make sure `config.is_decoder=False` for "
-                "bi-directional self-attention."
-            )
+        if not config.is_decoder:
+            logger.warning("If you want to use `TFBertLMHeadModel` as a standalone, add `is_decoder=True.`")
 
-        self.bert = TFBertMainLayer(config, name="bert")
-        self.mlm = TFBertMLMHead(config, self.bert.embeddings, name="mlm___cls")
+        self.initializer_range = config.initializer_range
+        self.bert = TFBertMainLayer(config=config, add_pooling_layer=False, name="bert")
+        self.cls = TFBertOnlyMLMHead(config=config, name="cls")
 
     def get_output_embeddings(self):
         return self.bert.embeddings
 
-    @add_start_docstrings_to_callable(BERT_INPUTS_DOCSTRING.format("batch_size, sequence_length"))
+    def resize_token_embeddings(self, new_num_tokens):
+        super().resize_token_embeddings(new_num_tokens=new_num_tokens)
+
+        if new_num_tokens is not None:
+            self.cls.predictions.decoder = tf.keras.layers.Dense(
+                new_num_tokens,
+                kernel_initializer=get_initializer(self.initializer_range),
+                use_bias=False,
+                name="decoder",
+            )
+
     @add_code_sample_docstrings(
         tokenizer_class=_TOKENIZER_FOR_DOC,
         checkpoint="bert-base-cased",
-        output_type=TFMaskedLMOutput,
+        output_type=TFCausalLMOutput,
         config_class=_CONFIG_FOR_DOC,
     )
     def call(
         self,
-        inputs=None,
+        input_ids=None,
         attention_mask=None,
         token_type_ids=None,
         position_ids=None,
         head_mask=None,
         inputs_embeds=None,
         output_attentions=None,
         output_hidden_states=None,
-        return_dict=None,
+        return_dict=True,
         labels=None,
         training=False,
     ):
         r"""
         labels (:obj:`tf.Tensor` of shape :obj:`(batch_size, sequence_length)`, `optional`):
-            Labels for computing the masked language modeling loss.
-            Indices should be in ``[-100, 0, ..., config.vocab_size]`` (see ``input_ids`` docstring)
-            Tokens with indices set to ``-100`` are ignored (masked), the loss is only computed for the tokens with labels
-            in ``[0, ..., config.vocab_size]``
+            Labels for computing the cross entropy classification loss.
```

Comment:
```suggestion
            Labels for computing the language modeling loss.
```
This is not a classification task.